provisioner/provision.js

```diff
@@ -79,61 +80,90 @@ var stats = {
   failure: 0,
 };
 
+// For when you want to be really certain that the program will
+// exit
+function exitTimer(time) {
+  var t = time || 30000;
+  setTimeout(() => {
+    debug('hey, so you probably are trying to figure out');
+    debug('why this process suddenly disappeared.  a major');
+    debug('error occured and you only get %d ms to exit after', t);
+    debug('before we process.exit(44). handle things faster!');
+    process.exit(44);
+  }, t);
+}
+
 /**
  * Start running a provisioner.
  */
 Provisioner.prototype.run = function () {
-  var that = this;
-
   this.__keepRunning = true;
+
   this.__watchDog.on('expired', function () {
-    debug('[alert-operator] Provisioner is stuck, killing');
+    debug('[alert-operator] provisioning iteration exceeded max time');
+    exitTimer(MAX_KILL_TIME);
     throw new Error('WatchDog expired');
   });
+
   this.__watchDog.start();
 
-  function provisionIteration () {
+  var provisionIteration = async () => {
     debug('starting iteration %d, successes %d, failures %d',
-          stats.iterations++,
+          stats.iterations,
           stats.success,
           stats.failure);
 
-    that.__watchDog.touch();
+    this.__watchDog.touch();
 
     // We should make sure that we're not just permanently failing
     // We also don't want to
-    if (stats.iterations > 20 && stats.failures > stats.success) {
-      debug('[alert-operator] killing provisioner because it has run ' +
-            'for a while but has failed lots of iterations');
+    if (stats.failure > MAX_FAILURES) {
```

Comment:
My original plan was to do something about failure percentages and all that.  I think consecutive errors is enough